### DISCUSSION TITLE
Does GraphSAGE support weighted graph?

### DISCUSSION THREAD

**Discussion by scottshufe** [pyg-team/pytorch_geometric#5983] [Q&A] [answered]
Does graphsage support weighted graph?

**Comment by wsad1** [comment, marked as answer]
No `GraphSage` in pyg doesn't support edge weight (https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/models/basic_gnn.html#GraphSAGE).

**Reply by scottshufe** [reply]
Could you please give me some detailed explanation?
And does the original GraphSAGE algorithm support weighted graph?

**Reply by wsad1** [reply]
The original `GraphSage` paper also doesn't support edge weights. From the documentation of [GraphSage](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.SAGEConv) it only uses node embeddings while message passing, edge weight is not used. There are other GNN layer like [GCN](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GCNConv) and [GAT](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GATConv) that support edge attribute/weights.

**Reply by scottshufe** [reply]
Thanks for your reply!

**Reply by rusty1s** [reply]
For `SAGEConv` with `edge_weight` support, take a look at `GraphConv`.

**Reply by scottshufe** [reply]
Thanks, I have found the answer from this issue: [https://github.com/pyg-team/pytorch_geometric/issues/1961](url)

**Reply by harirRzi** [reply]
Hello. I have multiple graphs and I want to use the edge weights for each of the graphs. I want to train different Graph neural network models with part of those multiple graphs and then test on the rest of graphs to classify nodes. I did this with GCN model, which supports edge weights and the prediction was good. Now I want to use GraphSage to do it but seems it doesn't support edge_weight. I am wondering if GraphSAGE is similar to GraphConv you mentioned? Can you show me an example for GraphConv if it is not similar to GCN?

**Reply by rusty1s** [reply]
Yes, `GraphConv` is mostly equivalent to `SAGEConv`. `GraphConv` has `aggr="sum"` by default (and supports edge weights), while `SAGEConv` uses `aggr="mean"` by default.
